packages/next/server/render.tsx

```diff
@@ -1373,11 +1374,13 @@ export async function renderToHTML(
           enhanceComponents(options, App, Component)
 
         if (renderShell) {
-          return renderShell(EnhancedApp, EnhancedComponent).then(() => {
-            // When using concurrent features, we don't have or need the full
-            // html so it's fine to return nothing here.
-            return { html: '', head }
-          })
+          return renderShell(EnhancedApp, EnhancedComponent).then(
+            async (stream) => {
+              await stream.allReady
+              const html = await streamToString(stream)
+              return { html, head }
```

Comment:
This means that we won't have streaming. Can we just return the shell as html here instead of the full stream?